### PR TITLE
COMMERCE-11178 Generic Payment Gateway Service Layer

### PR DESCRIPTION
**Pull request by luca-pellizzon**
[COMMERCE-11178](https://issues.liferay.com/browse/COMMERCE-11178)

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/upgrade/v1_4_0/CommercePaymentEntryUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+		_resourceActionLocalService = resourceActionLocalService;
+	}
+
+	@Override
+	public void doUpgrade() throws Exception {
+		CommercePaymentEntryTable.create();
+		CommercePaymentEntryAuditTable.create();
+
+		_resourceActionLocalService.checkResourceActions(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/upgrade/v1_4_0/CommercePaymentEntryUpgradeProcess.java:41]: Don't we need to do the same for audit?

File: modules/apps/commerce/commerce-payment-service/src/main/resources/resource-actions/default.xml [hunk 2]
```diff
@@ -2,6 +2,33 @@
...
+				<action-key>VIEW</action-key>
+			</guest-unsupported>
+			<owner-defaults>
+				<action-key>DELETE</action-key>
+				<action-key>PERMISSIONS</action-key>
+				<action-key>UPDATE</action-key>
+				<action-key>VIEW</action-key>
+			</owner-defaults>
+		</permissions>
+	</model-resource>
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-payment-service/src/main/resources/resource-actions/default.xml:31]: Don't we need to do the same for audit?

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@luca-pellizzon please check my comments, thank you!

File: modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentEntryAuditLocalServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	property = "model.class.name=com.liferay.commerce.payment.model.CommercePaymentEntryAudit",
+	service = AopService.class
+)
+public class CommercePaymentEntryAuditLocalServiceImpl
+	extends CommercePaymentEntryAuditLocalServiceBaseImpl {
+
+	@Override
+	public CommercePaymentEntryAudit addCommercePaymentEntryAudit(
+			long userId, long commercePaymentEntryId, BigDecimal amount,
+			String currencyCode, String logType, String logTypeSettings)
```
[the remark this answers is not in the document]
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentEntryAuditLocalServiceImpl.java]: they are not UnicodeProperties, they are like logTypeSettings of the CommerceInventoryAudit

**Comment by luca-pellizzon** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-payment-service/service.xml [hunk 4]
```diff
@@ -4,6 +4,79 @@
...
+
+		<!-- Other fields -->
+
+		<column name="classNameId" type="long" />
+		<column name="classPK" type="long" />
+		<column name="amount" type="BigDecimal" />
+		<column name="currencyCode" type="String" />
+		<column name="paymentMethodName" type="String" />
+		<column name="transactionCode" type="String" />
+		<column name="paymentStatus" type="int" />
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-payment-service/service.xml:29]: Sort alphabetically: paymentStatus before transactionCode

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@luca-pellizzon just one small thing, then it's okay for me. Thank you!

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@luca-pellizzon Please check my comments, thanks!

**Comment by luca-pellizzon** [comment]
ci:stop

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Review by alessiorendina — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:stop

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:stop

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:forward

**Pull request closed without merging**